xbmc/cores/VideoPlayer/VideoPlayer.h

```diff
@@ -168,27 +168,20 @@ class CCurrentStream
   }
 };
 
-typedef struct SelectionStream
+class SelectionStream : public DemuxSelectInfo
```

Comment:
I want SelectionStream to be rewritten. Note that DemuxStream is also used by paplayer that does not need all this. DemuxSelectInfo should not be part of CDemuxStream. Everything related so selection has to be inside this SelectionStream class.